pageserver/src/walreceiver/walreceiver_connection.rs

```diff
@@ -70,10 +71,19 @@ pub async fn handle_walreceiver_connection(
         let mut config = wal_source_connconf.to_tokio_postgres_config();
         config.application_name("pageserver");
         config.replication_mode(tokio_postgres::config::ReplicationMode::Physical);
-        time::timeout(connect_timeout, config.connect(postgres::NoTls))
-            .await
-            .context("Timed out while waiting for walreceiver connection to open")?
-            .context("Failed to open walreceiver connection")?
+        match time::timeout(connect_timeout, config.connect(postgres::NoTls)).await {
+            Ok(other_res) => match other_res {
```

Comment:
what is the meaning of other here ?

Reply:
Reworded and changed the match a bit, to improve readability.